pkg/controller/backup/backup_controller.go

```diff
@@ -263,3 +256,348 @@ func (c *Controller) enqueueBackup(obj interface{}) {
 	}
 	c.queue.Add(key)
 }
+
+// detectBackupJobOrPodFailure detect backup job or pod failure.
+// it will record failure info to backup status, it is to realize reentrant logic for spec.backoffRetryPolicy. so it only record snapshot backup failed now.
+func (c *Controller) detectBackupJobOrPodFailure(backup *v1alpha1.Backup) (bool, string, error) {
```

Comment:
if detectBackupJobOrPodFailure , how to get the failled log when meet error?

Reply:
the fialed log will be recorded in `status.backoffRetryStatus`, it is a array of `BackoffRetryRecord`, and there is a field `retryReason` is to record this message. but i don't record pod/job's message or reason, because i find it just `Error` when it failed. but add the original failed message is better, i will add them in `retryReason`.

Reply:
I think about it for a moment, and record the original error job or pod to `BackoffRetryRecord`'s `originalReason`, In this way, it's easy to implement an interface which can distinguish specific errors and decide whether to retry in the future.